### DISCUSSION TITLE
Is it possible to replace rg / ripgrep with rga / ripgrep-all ?

### DISCUSSION THREAD

**Discussion by onjour** [sxyazi/yazi#1084] [1. Q&A]
I'm not very tech savvy and I work with a lot of different files — including many PDFs.

I love _yazi_ and one thing I'm missing so far is the ability to search inside PDFs for text. 

I found that a [modified _rg_ exists](https://github.com/phiresky/ripgrep-all) specifically for this — but also many other file formats E-Books, Office documents, zip, tar.gz — and I've been wondering if it would be possible to replace the _rg_ search bound to the 'S' key in yazi to the _rga_ command that would better suit my needs? 

That would be great! If not, can someone help me to achieve PDF searching capability with the tools offered by default?

Thank you!

**Comment by lpnh** [comment]
@DreamMaoMao made this [awesome plugin](https://github.com/DreamMaoMao/fg.yazi) for searching using `rg` with `fzf` preview. I've included `rga` on my [fork](https://github.com/lpnh/fg.yazi/tree/rga). Maybe you could ask them if they would be interested in adding support to `rga`? If so, I'd be happy to send a PR to merge it. In the meantime, feel free to use and test my fork by using the `rga` branch like this:

```
git clone -b rga https://github.com/lpnh/fg.yazi.git ~/.config/yazi/plugins/fg.yazi
```

And make sure to add the keymap to it in the `keymap.toml` file, like this:

```
[[manager.prepend_keymap]]
on   = [ "f","a" ]
run  = "plugin fg --args='rga'"
desc = "find file by content (ripgrep-all)"
```

**Reply by DreamMaoMao** [reply]
That's a good idea,but how to preview  pdf file?

**Reply by lpnh** [reply]
This is a good question. To be honest, I hadn't thought about it yet. I kept the preview because the "regular" files can still benefit from it, but it's not as useful for PDFs, I guess.

**Comment by sxyazi** [comment]
It's now supported in https://github.com/sxyazi/yazi/pull/2383, thanks @paperbenni 🎉🎉
